### PR TITLE
Add support to pgvector as vector store.

### PR DESCRIPTION
**Pull request by ProGM**
Trying to implement #41 

Any suggestion is welcome.

Not sure if the pipeline will work ^^'

(cc @andreibondarev )

### CODE REVIEW

**Comment by andreibondarev** [comment]
@ProGM Would you please fix the failing `standardrb` check?
```
standardrb --fix
```
should suffice.

File: spec/vectorsearch/pgvector_spec.rb [hunk 1]
```diff
@@ -0,0 +1,144 @@
+# frozen_string_literal: true
+
+require "pg"
+
+if ENV['POSTGRES_URL']
```
Comment (andreibondarev) [inline spec/vectorsearch/pgvector_spec.rb:5]: Could you please add this env var to `.env.example`?

File: langchain.gemspec [hunk 2]
```diff
@@ -42,7 +42,9 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency "hugging-face", "~> 0.3.4"
   spec.add_development_dependency "milvus", "~> 0.9.0"
   spec.add_development_dependency "nokogiri", "~> 1.13"
-  spec.add_development_dependency "pdf-reader", "~> 1.4"
+  spec.add_development_dependency "pg", "~> 1.5"
+  spec.add_development_dependency "pgvector", "~> 0.2"
+  spec.add_development_dependency "pdf-reader", "~> 1.3"
```
Comment (andreibondarev) [inline langchain.gemspec:47]: Did we need to downgrade the `pdf_reader` from `~> 1.4` to `~> 1.3`?

**Comment by andreibondarev** [comment]
@ProGM This works great! I ran the tests locally, and tested out the instance methods.

Any chance you could add Rdoc-style documentation to describe what's happening?

**Comment by ProGM** [comment]
@andreibondarev Done! Please take a look if this works for you :)

File: langchain.gemspec [hunk 3]
```diff
@@ -42,6 +42,8 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency "hugging-face", "~> 0.3.4"
   spec.add_development_dependency "milvus", "~> 0.9.0"
   spec.add_development_dependency "nokogiri", "~> 1.13"
+  spec.add_development_dependency "pg", "~> 1.5"
+  spec.add_development_dependency "pgvector", "~> 0.2" if RUBY_VERSION >= "3.0.0"
```
[the remark this answers is not in the document]
Comment (andreibondarev) [reply langchain.gemspec:46]: Oh -- I see that the support to Ruby < 3 was dropped here: https://github.com/pgvector/pgvector-ruby/commit/670cfbbefb3dda5239ef98dc1c5aeecdb16294bb

[the remark this answers is not in the document]
Comment (ProGM) [reply langchain.gemspec:46]: Unfortunately yes :(

To be honest, ruby 2.7 is in EOL: https://endoflife.date/ruby

Considering that this is a pretty new/innovative gem, you can also consider to drop support for it.

[the remark this answers is not in the document]
Comment (andreibondarev) [reply langchain.gemspec:46]: Do you anticipate any issues with potentially downgrading pgvector to 1.x to continue supporting Ruby 2.7?

I need to think a bit about dropping 2.7 support. Although most likely it's all good.

[the remark this answers is not in the document]
Comment (ProGM) [reply langchain.gemspec:46]: No idea of potential issues... I'll take a look to the CHANGELOG ^^

**Comment by ProGM** [comment]
@andreibondarev I locked pgvector to < 0.2, since there's no actual code change between the two versions. In this way, it should work with ruby 2.7. I think it should fix for now.

What about the table indexing?

**Comment by andreibondarev** [comment]
@ProGM This is excellent, thank you!! 🎉 

Would you mind spinning up a separate Issue for table indexing? I def think we should tackle it.

**Comment by andreibondarev** [comment]
Quick follow-up this to your PR: https://github.com/andreibondarev/langchainrb/pull/65

**Pull request merged**